synapseformation/schemas/long.schema.yml

```diff
@@ -0,0 +1,155 @@
+$base: "https://synapse.org#"
+
+$namespaces:
+  syn: "https://synapse.org#"
+
+$graph:
+
+- name: RootDoc
+  type: documentation
+  doc:
+    - |
+      # SynapseFormation Long Template
+
+      TBD
+      
+
+- name: ResourceRecord
+  doc: |
+    The abstract base type for a Synapse resource.
+  type: record
+  abstract: true
+  fields:
+  - name: name
+    doc: ""
+    type: string?
+  - name: id
+    doc: "Synapse ID"
+    type: string?
+
+- name: EntityRecord
+  doc: |
+    The abstract base type for a Synapse entity.
+  type: record
+  extends: ResourceRecord
+  abstract: true
+  fields:
+  - name: children
+    doc: ""
+    type: FolderRecord[]?
+  - name: acl
+    doc: ""
+    type: AccessControlListRecord[]?
+
+- name: ProjectRecord
+  doc: |
+    The type for a Synapse project.
+  type: record
+  extends: EntityRecord
+  documentRoot: true
+  docParent: "#RootDoc"
+  fields:
+  - name: type
+    doc: ""
+    jsonldPredicate: syn:type
+    type:
+      type: enum
+      symbols:
+      - Project
+
+- name: FolderRecord
+  doc: |
+    The type for a Synapse folder.
+  type: record
+  extends: EntityRecord
+  docParent: "#RootDoc"
+  fields:
+  - name: type
+    doc: ""
+    jsonldPredicate: syn:type
+    type:
+      type: enum
+      symbols:
+      - Folder
+
+- name: AccessType
```

Comment:
I wonder if we should split up the "EntityAccessType" and "QueueAccessType" for now, because technically some of these values are only allowed for Entity or Queues.

Reply:
Good idea!